pkg/api/types.go

```diff
@@ -2167,19 +2179,32 @@ func (k *KubernetesConfig) RequiresDocker() bool {
 }
 
 // SetCloudProviderBackoffDefaults sets default cloudprovider backoff config
-func (k *KubernetesConfig) SetCloudProviderBackoffDefaults() {
+func (p *Properties) SetCloudProviderBackoffDefaults() {
+	k := p.OrchestratorProfile.KubernetesConfig
 	if k.CloudProviderBackoffDuration == 0 {
 		k.CloudProviderBackoffDuration = DefaultKubernetesCloudProviderBackoffDuration
+		if p.IsAzureStackCloud() {
+			k.CloudProviderBackoffDuration = DefaultAzureStackKubernetesCloudProviderBackoffDuration
+		}
 	}
 	if k.CloudProviderBackoffRetries == 0 {
 		k.CloudProviderBackoffRetries = DefaultKubernetesCloudProviderBackoffRetries
+		if p.IsAzureStackCloud() {
+			k.CloudProviderBackoffRetries = DefaultAzureStackKubernetesCloudProviderBackoffRetries
+		}
 	}
 	if k.CloudProviderBackoffMode != CloudProviderBackoffModeV2 {
 		if k.CloudProviderBackoffExponent == 0 {
 			k.CloudProviderBackoffExponent = DefaultKubernetesCloudProviderBackoffExponent
+			if p.IsAzureStackCloud() {
```

Comment:
This can be removed.

Reply:
I would rather keep it even if regular defaults and large-cluster defaults match.
Regular defaults can then evolve independently.